eras/conway/impl/src/Cardano/Ledger/Conway/Rules/GovCert.hs

```diff
@@ -159,9 +172,23 @@ conwayGovCertTransition = do
     ConwayResignCommitteeColdKey coldK -> do
       checkColdKeyHasNotResigned coldK vsCommitteeHotKeys
       pure $ vState {vsCommitteeHotKeys = Map.insert coldK Nothing vsCommitteeHotKeys}
-    ConwayUpdateDRep cred _mAnchor -> do
+    ConwayUpdateDRep cred mAnchor -> do
       Map.notMember cred vsDReps ?! ConwayDRepNotRegistered cred
-      pure vState -- TODO: update anchor
+      epochNo <- liftSTS $ do
+        ei <- asks epochInfoPure
+        epochInfoEpoch ei cgceCurrentSlot
+      let newTimeout = epochNo + cgcePParams ^. ppDRepActivityL
+      pure $
+        vState
+          & vsDRepsL
+            %~ Map.update
+              ( \x ->
+                  Just $
+                    x
+                      & drepExpiryL .~ newTimeout
+                      & drepAnchorL .~ mAnchor
+              )
+              cred
```

Comment:
It looks like the `drepExpiry` is supposed to be the `Epoch` in which the DRep should expire. But during registration (a few lines above) the `drepExpiry` field is considererd to be the "duration" of epochs that the drep should not expire for.

I may be wrong, but I think that the drep expiration calculation with respect to the current epoch should happen during registration? And here only the anchor should be updated? If `ConwayUpdateDRep` is indeed intended for expiry updates as well, then the `drepActivity` duration should be added yet again to the existing expiry epochNo here?